scripts/gis-material-update/fetch/fetch_data.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+config_yaml="/haitaton-gis/config.yaml"
+
+# Parse configuration YAML
+# Built using:
+# yq --version
+# yq (https://github.com/mikefarah/yq/) version 4.16.2
+
+data_object=$1
+
+parse_config () {
+    parsed_value=$(yq eval -e ".${1}" - < $config_yaml) > /dev/null 2>&1
+    if [ $? -eq 0 ]; then
+        retval="$parsed_value"
+    else
+        retval=""
+    fi
+    echo "$retval"
+}
+
+# dynamic configuration
+# refer to configuration YAML
+
+cfg_addr () {
+    echo "${data_object}.addr"
+}
+
+cfg_local_file () {
+    echo "${data_object}.local_file"
+}
+
+cfg_layer () {
+    echo "${data_object}.layer"
+}
+
+# download path
+download_dir=$(parse_config "common.download_path")
+
+# if we have progressed this far, object is supported in configuration
+# now it remains to find out how to download files
+addr=$(parse_config $(cfg_addr $data_object))
+layer=$(parse_config $(cfg_layer $data_object))
+local_file=${download_dir}/$(parse_config $(cfg_local_file $data_object))
+
+case $data_object in
+hsl|osm)
+    echo wget -O "$local_file $addr"
+    wget -O "$local_file" "$addr"
+    ;;
+ylre_katualue|ylre_katuosat|maka_autoliikennemaarat)
```

Comment:
Maybe the method for fetching the data could be a part of the config?

Like if the config was something like:
```
ylre_katualue:
  addr: "https://kartta.hel.fi/ws/geoserver/avoindata/wfs"
  layer: "avoindata:YLRE_Katualue_alue"
  local_file: "ylre_katualue_alue.gpkg"
  method: ogr2ogr
```
This could match that value:
```
ogr2ogr)
```

Then this script wouldn't have to know about each individual data source.

Reply:
Nice suggestion. Currently only WFS is used (for all materials here), but later on I'd expect ogr2ogr usage with other drivers also (shapefile, Geopackage, ...). I would also expect fine grained layer options, depending on actual material. Due to this I'd still keep the "material" approach, even though it requires updating of scripts.